core/trino-main/src/main/java/io/trino/server/security/ResourceAccessType.java

```diff
@@ -46,18 +46,14 @@ public AccessType getAccessType(ResourceInfo resourceInfo)
             if (accessType.isPresent()) {
                 return accessType.get();
             }
-            // check if the resource class has an access type declared for all methods
-            accessType = resourceAccessTypeLoader.getAccessType(resourceInfo.getResourceClass());
-            if (accessType.isPresent()) {
-                return accessType.get();
-            }
-            // in some cases there the resource is a nested class, so check the parent class
-            // we currently only check one level, but we could handle multiple nesting levels if necessary
-            if (resourceInfo.getResourceClass().getDeclaringClass() != null) {
-                accessType = resourceAccessTypeLoader.getAccessType(resourceInfo.getResourceClass().getDeclaringClass());
+            // check if the resource class or it's parents has an access type declared for all methods
+            Class<?> current = resourceInfo.getResourceClass();
+            while (current != null) {
+                accessType = resourceAccessTypeLoader.getAccessType(current);
                 if (accessType.isPresent()) {
                     return accessType.get();
                 }
+                current = current.getDeclaringClass();
```

Comment:
Reintroduce comment about it is for handlint nested classes.

Reply:
Reintroduced